src/ui/components/Library/Team/View/TestRuns/Overview/RunResults.tsx

```diff
@@ -11,31 +11,33 @@ import styles from "../../../../Library.module.css";
 export function RunResults() {
   const testRun = useContext(TestRunOverviewContext).testRun!;
 
-  const sortedRecordings = orderBy(testRun.recordings, "date", "desc");
-  const passedRecordings = sortedRecordings.filter(r => r.metadata?.test?.result === "passed");
-  const failedRecordings = sortedRecordings.filter(
-    r => r.metadata?.test?.result && ["failed", "timedOut"].includes(r.metadata?.test?.result)
+  const { passedRecordings, failedRecordings, flakyRecordings } = useMemo(
+    () => groupRecordings(testRun.recordings),
+    [testRun.recordings]
```

Comment:
`groupRecordings` isn't super expensive, but it's doing a lot of iterating. Seems like it would be nice if we only called it once, higher up, and passed the values down (rather than calling it in `RunResults` and `RunStats`– and also we'll need to call it in `TestRunListItem` too probably)